src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -36,7 +36,24 @@ public Task<Geoposition> GetGeopositionAsync()
 		public Task<Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
 			=> GetGeopositionAsync();
 
-		public static async Task<GeolocationAccessStatus> RequestAccessAsync() => GeolocationAccessStatus.Allowed;
+		public static async Task<GeolocationAccessStatus> RequestAccessAsync()
+		{
+			// below 6.0 (API 23), permission are granted
+			if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.M)
+			{
+				return GeolocationAccessStatus.Allowed;
+			}
+
+			// check if permission is granted
+			if (Android.Support.V4.Content.ContextCompat.CheckSelfPermission(Uno.UI.ContextHelper.Current, Android.Manifest.Permission.AccessFineLocation)
```

Comment:
Should AccessCoarseLocation be considered also?

Reply:
Probably, but I have no idea how to really convert between permission system on UWP and on Android. On Windows, we have requestprecision in meters; on Android, only coarse/fine (and I found no article with any precise info what is coarse, and what is fine).
And on UWP you can RequestAccess, than set requested precision. If RequestAsync ask system about coarse, than app would not be able to change to fine...